llvm/lib/CodeGen/SelectionDAG/LegalizeFloatTypes.cpp

```diff
@@ -2999,8 +3003,10 @@ SDValue DAGTypeLegalizer::PromoteFloatRes_UnaryOp(SDNode *N) {
   EVT VT = N->getValueType(0);
   EVT NVT = TLI.getTypeToTransformTo(*DAG.getContext(), VT);
   SDValue Op = GetPromotedFloat(N->getOperand(0));
-
-  return DAG.getNode(N->getOpcode(), SDLoc(N), NVT, Op);
+  if (N->getOpcode() == ISD::AssertNoFPClass)
+    return DAG.getNode(N->getOpcode(), SDLoc(N), NVT, Op, N->getOperand(1));
+  else
+    return DAG.getNode(N->getOpcode(), SDLoc(N), NVT, Op);
```

Comment:
No else after return. This doesn't really fall under PromoteFloatRes_UnaryOp, I would just move this into its own legalization function